### PR TITLE
feat(aapd-1534): route users from BYS to v2 HAS appeal form

### PR DESCRIPTION
**Pull request by bryn-pins**
## Ticket Number

https://pins-ds.atlassian.net/browse/AAPD-1534

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/Planning-Inspectorate/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/forms-web-app/src/dynamic-forms/middleware/get-journey-response-for-appellant.js [hunk 1]
```diff
@@ -32,7 +34,9 @@ module.exports = async (request, response, next) => {
 		logger.error(error);
 	}
 
-	if (!submission) return response.status(404).render('error/not-found');
+	if (!submission || !(await isFeatureActive(FLAG.APPEAL_FORM_V2, submission.LPACode))) {
```
Comment (bryn-pins) [inline packages/forms-web-app/src/dynamic-forms/middleware/get-journey-response-for-appellant.js:37]: replacing the middleware check with a check here so we don't need to keep the old req.session.appeal data

File: packages/forms-web-app/src/routes/appeals/index.js [hunk 2]
```diff
@@ -6,25 +6,12 @@ const yourAppealsRouter = require('./your-appeals/index');
...
 const selectedAppealRouter = require('./selected-appeal/selected-appeal');
 const dynamicSubmission = require('../appellant-submission/submission-form');
-const { isFeatureActive } = require('../../featureFlag');
-const { FLAG } = require('@pins/common/src/feature-flags');
 
 router.use('/your-appeals', yourAppealsRouter);
 router.get('/no-appeals', noAppealsController.get);
 
 // householder appeals
-router.use(
```
Comment (bryn-pins) [inline packages/forms-web-app/src/routes/appeals/index.js:16]: check handled in get-journey-response-for-appellant.js

File: packages/forms-web-app/src/controllers/appeal-householder-decision/list-of-documents.js [hunk 3]
```diff
@@ -11,8 +20,31 @@ const postListOfDocuments = async (req, res) => {
 
 	try {
 		if (req.body['save-and-return'] !== '') {
-			return res.redirect('/appeal-householder-decision/task-list');
+			const appeal = req.session.appeal;
+
+			const usingV2Form = await isFeatureActive(FLAG.HAS_QUESTIONNAIRE, appeal.lpaCode);
```
Comment (topickering) [inline packages/forms-web-app/src/controllers/appeal-householder-decision/list-of-documents.js:25]: Is this the right feature flag, or should it be APPEAL_FORM_V2?

**Review by topickering — APPROVED** [review]
LGTM

**Pull request merged**